packages/@aws-cdk/aws-ecr-assets/lib/image-asset.ts

```diff
@@ -445,6 +477,14 @@ function validateBuildArgs(buildArgs?: { [key: string]: string }) {
   }
 }
 
+function validateBuildSecrets(buildSecrets?: { [key: string]: string }) {
+  for (const [key, value] of Object.entries(buildSecrets || {})) {
+    if (Token.isUnresolved(key) || Token.isUnresolved(value)) {
+      throw new Error('Cannot use tokens in keys or values of "buildSecrets" since they are needed before deployment');
+    }
+  }
+}
```

Comment:
It's great that we're doing validation, but it would be nice to combine this function with `validateBuildArgs`. How about 

```
function validateBuildProps(buildPropName: string, buildProps?: { [key: string]: string }) {
  for (const [key, value] of Object.entries(buildArgs || {})) {
    if (Token.isUnresolved(key) || Token.isUnresolved(value)) {
      throw new Error(`Cannot use tokens in keys or values of '${buildPropName}' since they are needed before deployment`);
    }
  }
}
```

then `validateBuildSecrets()` and `validateBuildArgs` become:
```
function validateBuildArgs(buildArgs?: { [key: string]: string }) {
  vadliateBuildProps('buildArgs', props.buildArgs);
}
``` 

```
function validateBuildSecrets(buildSecrets?: { [key: string]: string }) {
  vadliateBuildProps('buildSecrets', props.buildSecrets);
}
```

Reply:
Good shout, done.